mindsdb/integrations/handlers/huggingface_api_handler/huggingface_api_handler.py

```diff
@@ -16,118 +17,166 @@ class HuggingFaceInferenceAPIHandler(BaseMLEngine):
+        elif args['task'] == 'question-answering':
             nlp = NLP(api_key)
             result_df = nlp.question_answering_in_df(
                 df,
-                args['using']['question_column'],
-                args['using']['context_column'],
-                args['using']['model'] if 'model' in args['using'] else None
+                input_column,
+                args['context_column'],
+                model_name
             )
 
-        elif args['using']['task'] == 'sentence-similarity':
+        elif args['task'] == 'sentence-similarity':
             nlp = NLP(api_key)
             result_df = nlp.sentence_similarity_in_df(
                 df,
-                args['using']['source_sentence_column'],
-                args['using']['sentence_column'],
-                args['using']['options'] if 'options' in args['using'] else None,
-                args['using']['model'] if 'model' in args['using'] else None
+                input_column,
+                args['sentence_column'],  # TODO name it input_column2?
+                args['options'] if 'options' in args else None,
+                model_name
             )
 
-        elif args['using']['task'] == 'zero-shot-classification':
+        elif args['task'] == 'zero-shot-classification':
             nlp = NLP(api_key)
             result_df = nlp.zero_shot_classification_in_df(
                 df,
-                args['using']['column'],
-                args['using']['candidate_labels'],
-                args['using']['parameters'] if 'parameters' in args['using'] else None,
-                args['using']['options'] if 'options' in args['using'] else None,
-                args['using']['model'] if 'model' in args['using'] else None
+                input_column,
+                args['candidate_labels'],
+                args['parameters'] if 'parameters' in args else None,
+                args['options'] if 'options' in args else None,
+                model_name
             )
 
-        elif args['using']['task'] == 'image-classification':
+        elif args['task'] == 'image-classification':
             cp = ComputerVision(api_key)
             result_df = cp.image_classification_in_df(
                 df,
-                args['using']['column'],
-                args['using']['model'] if 'model' in args['using'] else None
+                input_column,
+                model_name
             )
 
-        elif args['using']['task'] == 'object-detection':
+        elif args['task'] == 'object-detection':
             cp = ComputerVision(api_key)
             result_df = cp.object_detection_in_df(
                 df,
-                args['using']['column'],
-                args['using']['model'] if 'model' in args['using'] else None
+                input_column,
+                model_name
             )
 
-        elif args['using']['task'] == 'speech-recognition':
+        elif args['task'] == 'speech-recognition':
```

Comment:
I updated this to 'automatic-speech-recognition' in the package, because that is how `huggingface_hub` refers to it. We might need to change that here.